Makefile

```diff
@@ -55,6 +55,9 @@ CXXFLAGS  += -I$(LIBREM_PATH)/include
 CXXFLAGS  += -I$(SYSROOT)/local/include/rem -I$(SYSROOT)/include/rem
 CXXFLAGS  += $(EXTRA_CXXFLAGS)
 
+# XXX: GNU libc
+CPPFLAGS += -D_XOPEN_SOURCE=500
+
```

Comment:
I am not sure if I like this, it is quite an intrusive change? basically it means that
all .c files are now compiled with different flags.

Would it be possible to #define this before including limits.h instead?